src/uiSelectDirective.js

```diff
@@ -70,7 +70,18 @@ uis.directive('uiSelect',
 
         attrs.$observe('disabled', function() {
           // No need to use $eval() (thanks to ng-disabled) since we already get a boolean instead of a string
-          $select.disabled = attrs.disabled !== undefined ? attrs.disabled : false;
+          $select.disabledFlag = attrs.disabled !== undefined ? attrs.disabled : false;
+          $select.disabled = $select.disabledFlag || $select.disabledSelector;
+        });
+
+        // When inside a disabled fieldset, disabledWatch element will be disabled
+        var disabledWatch = angular.element("<input type='hidden' class='ui-select-disable-watch' aria-label=''></input>");
+        element.append(disabledWatch);
+
+        // Watch the disabled flag on disabledWatch, and set $select.disabled.
+        scope.$watch(function () { return disabledWatch.is(":disabled"); }, function (value) {
```

Comment:
.is() isn't part of [angular.element](https://docs.angularjs.org/api/ng/function/angular.element) so we'll be creating a dependency on jquery to make this work which we should avoid